### PR TITLE
Update docs to reflect affinity, taints, tolerations moving from alpha annotations to (beta) fields.

### PR DESCRIPTION
**Pull request by davidopp**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/2905)

### CODE REVIEW

**Comment by davidopp** [comment]
ref/ kubernetes/kubernetes#25320 kubernetes/kubernetes#25319

**Comment by davidopp** [comment]
ref/ #2737

File: docs/admin/daemons.md [hunk 1]
```diff
@@ -91,6 +88,14 @@ when the pod is created, so it is ignored by the scheduler).  Therefore:
    by the DaemonSet controller.
  - DaemonSet controller can make pods even when the scheduler has not been started, which can help cluster
    bootstrap.
+   
+Daemon pods do respect [taints and tolerations](/docs/user-guide/node-selection/index.md), but they are
```
Comment (gyliu513) [inline docs/admin/daemons.md]: How about use `/docs/user-guide/node-selection/index.md#taints-and-tolerations-beta-feature`

Comment (gyliu513) [inline docs/admin/daemons.md]: Some places using `Daemon pods` while some places using `DaemonSet pods`, shall we unify the name?

File: docs/admin/daemons.md [hunk 2]
```diff
@@ -91,6 +88,14 @@ when the pod is created, so it is ignored by the scheduler).  Therefore:
...
+created with tolerations for the `node.alpha.kubernetes.io/notReady` and `node.alpha.kubernetes.io/unreachable`
```
Comment (gyliu513) [inline docs/admin/daemons.md]: How about 

created with `NoExecute` tolerations for `node.alpha.kubernetes.io/unreachable` and `node.alpha.kubernetes.io/notReady` with no `tolerationSeconds`.

File: docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md [hunk 3]
```diff
@@ -48,10 +48,8 @@ It's enabled by default. It can be disabled:
 
 ### Marking add-on as critical
 
-To be critical an add-on has to run in `kube-system` namespace (configurable via flag)
-and have the following annotations specified:
-
-* `scheduler.alpha.kubernetes.io/critical-pod` set to empty string
-* `scheduler.alpha.kubernetes.io/tolerations` set to `[{"key":"CriticalAddonsOnly", "operator":"Exists"}]`
```
Comment (gyliu513) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: Updating here maybe more readable

PodSpec's `tolerations` field set to `[{"key":"CriticalAddonsOnly", "operator":"Exists"}]`

File: docs/admin/daemons.md (hunk 1 above)
Comment (davidopp) [inline docs/admin/daemons.md]: Done

Comment (davidopp) [inline docs/admin/daemons.md]: Changed the one instance of "daemonset pods" to "daemon pods"

File: docs/admin/daemons.md (hunk 2 above)
Comment (davidopp) [inline docs/admin/daemons.md]: Done

File: docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md (hunk 3 above)
Comment (davidopp) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: I don't understand -- how is what you suggested different from what I did?

Comment (gyliu513) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: The only difference is that modifying here can enable using short sentence and sequence based items which maybe more readable?

Comment (davidopp) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: Sorry, I still don't understand. Can you write down what text I currently have, and what text you are suggesting to replace it with?

Comment (gyliu513) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: Sorry for the confusion, how about as following?

To be critical an add-on has to run in `kube-system` namespace (configurable via flag)
and have the following annotations specified:

* `scheduler.alpha.kubernetes.io/critical-pod` set to empty string
* PodSpec's `tolerations` field set to `[{"key":"CriticalAddonsOnly", "operator":"Exists"}]`

Comment (davidopp) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: I think the current wording is a bit clearer in distinguishing annotation vs. field. They are very similar but I think for someone who is not familiar with the concepts, the version I wrote is more explicit and thus less likely to cause confusion.

Comment (gyliu513) [inline docs/concepts/cluster-administration/guaranteed-scheduling-critical-addon-pods.md]: got it, fair enough, thanks @davidopp

**Pull request merged**